<Type Name="ServerErrorException" FullName="Microsoft.Azure.Devices.Client.Exceptions.ServerErrorException">
  <TypeSignature Language="C#" Value="public sealed class ServerErrorException : Microsoft.Azure.Devices.Client.Exceptions.IotHubException" />
  <TypeSignature Language="ILAsm" Value=".class public auto ansi serializable sealed beforefieldinit ServerErrorException extends Microsoft.Azure.Devices.Client.Exceptions.IotHubException" />
  <TypeSignature Language="DocId" Value="T:Microsoft.Azure.Devices.Client.Exceptions.ServerErrorException" />
  <TypeSignature Language="VB.NET" Value="Public NotInheritable Class ServerErrorException&#xA;Inherits IotHubException" />
  <TypeSignature Language="F#" Value="type ServerErrorException = class&#xA;    inherit IotHubException" />
  <AssemblyInfo>
    <AssemblyName>Microsoft.Azure.Devices.Client</AssemblyName>
    <AssemblyVersion>1.31.2.0</AssemblyVersion>
    <AssemblyVersion>1.32.0.0</AssemblyVersion>
    <AssemblyVersion>1.36.0.0</AssemblyVersion>
    <AssemblyVersion>1.36.1.0</AssemblyVersion>
    <AssemblyVersion>1.36.6.0</AssemblyVersion>
    <AssemblyVersion>1.37.0.0</AssemblyVersion>
    <AssemblyVersion>1.37.1.0</AssemblyVersion>
    <AssemblyVersion>1.37.2.0</AssemblyVersion>
    <AssemblyVersion>1.38.0.0</AssemblyVersion>
    <AssemblyVersion>1.39.0.0</AssemblyVersion>
    <AssemblyVersion>1.40.0.0</AssemblyVersion>
    <AssemblyVersion>1.41.0.0</AssemblyVersion>
    <AssemblyVersion>1.41.1.0</AssemblyVersion>
    <AssemblyVersion>1.41.2.0</AssemblyVersion>
  </AssemblyInfo>
  <Base>
    <BaseTypeName>Microsoft.Azure.Devices.Client.Exceptions.IotHubException</BaseTypeName>
  </Base>
  <Interfaces />
  <Attributes>
    <Attribute>
      <AttributeName Language="C#">[System.Serializable]</AttributeName>
      <AttributeName Language="F#">[&lt;System.Serializable&gt;]</AttributeName>
    </Attribute>
  </Attributes>
  <Docs>
    <summary>
            The exception that is thrown when the IoT hub returned an internal service error.
            </summary>
    <remarks>
            This exception typically means the IoT hub service has encountered an unexpected error and is usually transient.
            Please review the
            <see href="https://docs.microsoft.com/azure/iot-hub/iot-hub-troubleshoot-error-500xxx-internal-errors">500xxx Internal errors</see>
            guide for more information. The best course of action is to retry your operation after some time. By default,
            the SDK will utilize the <see cref="T:Microsoft.Azure.Devices.Client.ExponentialBackoff" /> retry strategy.
            </remarks>
  </Docs>
  <Members>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public ServerErrorException ();" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor() cil managed" />
      <MemberSignature Language="DocId" Value="M:Microsoft.Azure.Devices.Client.Exceptions.ServerErrorException.#ctor" />
      <MemberSignature Language="VB.NET" Value="Public Sub New ()" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>Microsoft.Azure.Devices.Client</AssemblyName>
        <AssemblyVersion>1.31.2.0</AssemblyVersion>
        <AssemblyVersion>1.32.0.0</AssemblyVersion>
        <AssemblyVersion>1.36.0.0</AssemblyVersion>
        <AssemblyVersion>1.36.1.0</AssemblyVersion>
        <AssemblyVersion>1.36.6.0</AssemblyVersion>
        <AssemblyVersion>1.37.0.0</AssemblyVersion>
        <AssemblyVersion>1.37.1.0</AssemblyVersion>
        <AssemblyVersion>1.37.2.0</AssemblyVersion>
        <AssemblyVersion>1.38.0.0</AssemblyVersion>
        <AssemblyVersion>1.39.0.0</AssemblyVersion>
        <AssemblyVersion>1.40.0.0</AssemblyVersion>
        <AssemblyVersion>1.41.0.0</AssemblyVersion>
        <AssemblyVersion>1.41.1.0</AssemblyVersion>
        <AssemblyVersion>1.41.2.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters />
      <Docs>
        <summary>
            Initializes a new instance of the class.
            </summary>
        <remarks>To be added.</remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public ServerErrorException (string message);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message) cil managed" />
      <MemberSignature Language="DocId" Value="M:Microsoft.Azure.Devices.Client.Exceptions.ServerErrorException.#ctor(System.String)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String)" />
      <MemberSignature Language="F#" Value="new Microsoft.Azure.Devices.Client.Exceptions.ServerErrorException : string -&gt; Microsoft.Azure.Devices.Client.Exceptions.ServerErrorException" Usage="new Microsoft.Azure.Devices.Client.Exceptions.ServerErrorException message" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>Microsoft.Azure.Devices.Client</AssemblyName>
        <AssemblyVersion>1.31.2.0</AssemblyVersion>
        <AssemblyVersion>1.32.0.0</AssemblyVersion>
        <AssemblyVersion>1.36.0.0</AssemblyVersion>
        <AssemblyVersion>1.36.1.0</AssemblyVersion>
        <AssemblyVersion>1.36.6.0</AssemblyVersion>
        <AssemblyVersion>1.37.0.0</AssemblyVersion>
        <AssemblyVersion>1.37.1.0</AssemblyVersion>
        <AssemblyVersion>1.37.2.0</AssemblyVersion>
        <AssemblyVersion>1.38.0.0</AssemblyVersion>
        <AssemblyVersion>1.39.0.0</AssemblyVersion>
        <AssemblyVersion>1.40.0.0</AssemblyVersion>
        <AssemblyVersion>1.41.0.0</AssemblyVersion>
        <AssemblyVersion>1.41.1.0</AssemblyVersion>
        <AssemblyVersion>1.41.2.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
      </Parameters>
      <Docs>
        <param name="message">The error message.</param>
        <summary>
            Initializes a new instance of the class.
            </summary>
        <remarks>To be added.</remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public ServerErrorException (string message, Exception innerException);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message, class System.Exception innerException) cil managed" />
      <MemberSignature Language="DocId" Value="M:Microsoft.Azure.Devices.Client.Exceptions.ServerErrorException.#ctor(System.String,System.Exception)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String, innerException As Exception)" />
      <MemberSignature Language="F#" Value="new Microsoft.Azure.Devices.Client.Exceptions.ServerErrorException : string * Exception -&gt; Microsoft.Azure.Devices.Client.Exceptions.ServerErrorException" Usage="new Microsoft.Azure.Devices.Client.Exceptions.ServerErrorException (message, innerException)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>Microsoft.Azure.Devices.Client</AssemblyName>
        <AssemblyVersion>1.31.2.0</AssemblyVersion>
        <AssemblyVersion>1.32.0.0</AssemblyVersion>
        <AssemblyVersion>1.36.0.0</AssemblyVersion>
        <AssemblyVersion>1.36.1.0</AssemblyVersion>
        <AssemblyVersion>1.36.6.0</AssemblyVersion>
        <AssemblyVersion>1.37.0.0</AssemblyVersion>
        <AssemblyVersion>1.37.1.0</AssemblyVersion>
        <AssemblyVersion>1.37.2.0</AssemblyVersion>
        <AssemblyVersion>1.38.0.0</AssemblyVersion>
        <AssemblyVersion>1.39.0.0</AssemblyVersion>
        <AssemblyVersion>1.40.0.0</AssemblyVersion>
        <AssemblyVersion>1.41.0.0</AssemblyVersion>
        <AssemblyVersion>1.41.1.0</AssemblyVersion>
        <AssemblyVersion>1.41.2.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
        <Parameter Name="innerException" Type="System.Exception" />
      </Parameters>
      <Docs>
        <param name="message">The error message.</param>
        <param name="innerException">The exception instance that caused the current exception.</param>
        <summary>
            Initializes a new instance of the class.
            </summary>
        <remarks>To be added.</remarks>
      </Docs>
    </Member>
  </Members>
</Type>
